### PR TITLE
test(robot): add test case Test RWX Failover And Auto Salvage When Volume Is Faulted

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/9091

#### What this PR does / why we need it:

add test case Test RWX Failover And Auto Salvage When Volume Is Faulted

### CODE REVIEW

File: e2e/libs/metrics/metrics.py [hunk 1]
```diff
@@ -207,47 +207,53 @@ def check_longhorn_components_memory_cpu_usage():
             usage["backing-image-manager"]["memory"],
             cold_usage["backing-image-manager"]["memory"])
```
Comment (roger-ryao) [inline e2e/libs/metrics/metrics.py]:
```suggestion
            old_usage["backing-image-manager"]["memory"])
```
Typo

**Review by roger-ryao — CHANGES_REQUESTED** [review]

File: e2e/tests/regression/test_rwx_volume.robot [hunk 2]
```diff
@@ -137,35 +136,67 @@ Test ShareManager Status Current Image After Upgrade
...
+    And Setting auto-salvage is set to true
+    And Create storageclass longhorn-test with    numberOfReplicas=3    dataEngine=${DATA_ENGINE}
+    And Create statefulset 0     volume_type=RWX    sc_name=longhorn-test
+    And Wait for volume of statefulset 0 healthy
+    And Update volume of statefulset 0 replica count to 1
+    And Delete replica of statefulset 0 volume on all replica node
+    And Write 1 MB data to file data.bin in statefulset 0
+    And Get Longhorn components resource usage
+
+    When Power off volume node of statefulset 0
```
Comment (roger-ryao) [inline e2e/tests/regression/test_rwx_volume.robot:194]: If I remember correctly, test cases involving actions like `stopping kubelet` or `Power off` should be categorized under negative tests. Does this classification need to be adjusted?

File: e2e/libs/metrics/metrics.py (hunk 1 above)
Comment (yangchiu) [reply e2e/libs/metrics/metrics.py]: Updated. Thank you!

File: e2e/tests/regression/test_rwx_volume.robot (hunk 2 above)
Comment (yangchiu) [reply e2e/tests/regression/test_rwx_volume.robot:194]: I think it depends on whether we are` testing a corner case or intentionally injecting negative factors to test resiliency` or `verifying a behavior`. What do you think this test case should be categorized as? I am fine with either.

Comment (roger-ryao) [reply e2e/tests/regression/test_rwx_volume.robot:194]: Let's add the `node-down` tag to the case for now and leave the case location as is. Since this issue originated from an internal ticket, we can reorganize it based on the case tags.

Comment (yangchiu) [reply e2e/tests/regression/test_rwx_volume.robot:194]: Updated. Thank you.

**Review by roger-ryao — APPROVED** [review]
LGTM

**Comment by yangchiu** [comment]
@mergify backport v1.11.x
